### PR TITLE
Fix broken links on readme and docs homepage

### PR DESCRIPTION
**Pull request by nwmac**
Fixes broken links in the readme and on the docs landing page.

Extra fixes for: #11116

### CODE REVIEW

**Review by aalves08 — APPROVED** [review]
LGTM

Screenshot 2024-06-03 at 16 56 47
Screenshot 2024-06-03 at 16 56 49

File: README.md [hunk 1]
```diff
@@ -3,28 +3,25 @@
...
-For more information, see our [UI Walk-through documentation](https://rancher.github.io/dashboard/getting-started/ui-walkthrough).
+## Internal Developer Docs
+
+These docs are intended only for Dashboard UI developers.
 
-## Developer Docs
+If you're building extensions, please see: https://extensions.rancher.io.
 
-- [Rancher UI Devkit](https://rancher.github.io/dashboard/) - for our documentation home
-- [Developer Quick Start documentation](https://rancher.github.io/dashboard/getting-started/quickstart) - to get the Rancher Dashboard running
```
Comment (richard-cox) [inline README.md:22]: nit pick - might be useful to either keep a direct link to how to run the dashboard within the README (new URL is https://extensions.rancher.io/internal/getting-started/quickstart) or a clear comment the getting started guide is in the docs

File: README.md [hunk 2]
```diff
@@ -3,28 +3,25 @@
...
-
-Rancher Dashboard supports an extension mechanism that allows developers to independently provide additional functionality to Rancher. You can learn more from our [Rancher Extensions Docs](https://rancher.github.io/dashboard/extensions/introduction).
-
+Rancher Dashboard supports an extension mechanism that allows developers to independently provide additional functionality to Rancher. You can learn more from our [Rancher Extensions Docs](https://extensions.rancher.io).
 
 # What is it?
 
 Rancher Dashboard provides a sophisticated UI for managing Kubernetes clusters and Workloads.
 
 <img src="docusaurus/docs/getting-started/screenshots/home.png" />
```
Comment (richard-cox) [inline README.md:12]: this doesn't exist anymore

**Review by richard-cox — APPROVED** [review]
two comments, could be done later on?

**Pull request merged**